src/imap/scan_folders.rs

```diff
@@ -0,0 +1,75 @@
+use std::time::Instant;
+
+use crate::{config::Config, context::Context, dc_tools::time};
+use anyhow::Context as _;
+
+use crate::error::Result;
+use crate::imap::Imap;
+use async_std::prelude::*;
+
+use super::{get_folder_meaning, get_folder_meaning_by_name, FolderMeaning};
+
+impl Imap {
+    pub async fn scan_folders(&mut self, context: &Context) -> Result<()> {
+        use crate::config::Config::*;
+
+        // First of all, debounce to once per minute:
+        let mut last_scan = context.last_full_folder_scan.lock().await;
+        if let Some(last_scan) = *last_scan {
+            if last_scan.elapsed().as_secs() < 60 {
+                // For the first day after installation, we only debounce to 2s:
+
+                let configure_time = context.get_config_i64(Config::ConfiguredTimestamp).await;
+                if time() - configure_time > 24 * 60 * 60 || last_scan.elapsed().as_secs() < 2 {
+                    info!(context, "Not scanning, we recently already scanned");
+                    return Ok(());
+                }
+            }
+        }
+        info!(context, "Starting full folder scan");
+        last_scan.replace(Instant::now());
+
+        self.setup_handle(context).await?;
+        let session = self.session.as_mut();
+        let session = session.context("scan_folders(): IMAP No Connection established")?;
+        let folders: Vec<_> = session.list(Some(""), Some("*")).await?.collect().await;
+
+        for folder in folders {
+            let folder = folder?;
+            let foldername = folder.name();
+            info!(context, "Scanning folder: {}", foldername);
+
+            let folder_meaning = get_folder_meaning(&folder);
+            let folder_name_meaning = get_folder_meaning_by_name(&foldername);
+            // If there are two folders with the \Sent or \Spam flag, then the sent/spam folder will change all the time.
```

Comment:
Why should it change? It will change only if the order of folders are listed in different order each time, or for some other reason?

Reply:
It will be the first of the folders for only a very short time, but during this time `fetch_new_messages()` will be called, so that this actually matters (a little).

But if this comment only creates more confusion than it solves, I can remove it.

Reply:
See the comment below then, I think we need to set these folders once at the end of the loop.